svc/FetchService.ts

```diff
@@ -169,16 +216,20 @@ export class FetchService extends HoistService {
             Object.assign(defaultHeaders, isFunction(h) ? await h(opts) : h);
         }
 
-        return {
-            ...opts,
-            method,
-            headers: {
-                'Content-Type': isPost ? 'application/x-www-form-urlencoded' : 'text/plain',
-                ...defaultHeaders,
-                ...extraHeaders,
-                ...opts.headers
-            }
+        const headers = {
+            'Content-Type': isPost ? 'application/x-www-form-urlencoded' : 'text/plain',
+            ...defaultHeaders,
+            ...extraHeaders,
+            ...opts.headers
         };
+
+        const {correlationIdHeaderKey} = this;
+        if (opts.correlationId) {
+            throwIf(headers[correlationIdHeaderKey], 'Correlation ID already set via Fetch API.');
```

Comment:
I'd vote for warning here vs. throwing - does not seem like a critical enough conflict to warrant a throw. 

(Seems like there could be a chance that they are the same if a dev is doing something a bit weird/unnecessary). 

Error/warning message also seems a bit unclear to me - what about something like `Correlation ID already set in ${correlationIdHeaderKey} header within FetchOptions - ID specified directly at top level of FetchOptions will be ignored`. Case where FetchService is configured to auto-generate IDs might be worth calling out.

Reply:
Yeah - I'm assuming this will mainly trip up apps that already implemented their own support for correlation ID's by setting default headers. In that case, I think we'd want to be pretty loud. But point taken... we could warn here. I like your message change.  If we warn instead of throw, maybe we still prefer the `opts.correlationId` and write over the previously set header? Not sure which is more likely to be "correct" in that case.

Reply:
It's such an edge case I think it's low-stakes either way, but I liked how you had it respecting the header - that feels like a more granular / specific / advanced usage, and the opts id could have been auto-generated.